refactor: порт actions

vovakons opened the pull request:
## Что этот ПР делает

Порт actions с тг.
прок холдеры остались, поэтому оставил старый `spell_action`, надо будет потом перевести в нормальный `/datum/action/spell`

## Почему это хорошо для игры

Главный сказал надо.

## Демонстрация изменений

<img width="1229" height="1080" alt="image" src="https://github.com/user-attachments/assets/5cf838e9-5ab8-4e25-be19-febd52f06521" />


## Тестирование

Проверено на локалке, вроде багов не нашел, но лучше все таки проверить через тестмерж, потенциально создаст очень много багов.

vovakons committed:
create actions folder in datums and move action.dm

Создал папку code/datums/actions для всех экшнов.
Переместил файл action.dm в новую папку.

vovakons committed:
actions refactor in process

vovakons committed:
Merge branch 'master220' into actions-refactor

vovakons committed:
Merge branch 'master220' into actions-refactor

vovakons committed:
generic action pport

vovakons committed:
add basic actions

Добавлен item action.
Добавлен innate action.
Добавлен cooldown action.

vovakons committed:
fix errors

Исправление всех ошибок и первый пробный запуск.

vovakons committed:
items actions fix

Исправлены action у предметов.

vovakons committed:
item action bg fix

Исправление показа background текстуры для экшн кнопок.

vovakons committed:
internals action fix

vovakons committed:
spell action and magic spells actions

vovakons committed:
ninja actions fixes

Исправления action ниндзи.

littleboobs commented on code/datums/components/action_item_overlay.dm:
```diff
@@ -0,0 +1,79 @@
+		stack_trace("[type] created without a reference item or an item callback - one or the other is required.")
+		return COMPONENT_INCOMPATIBLE
+
+	src.item_ref = WEAKREF(item)
```
Мне кажется тут должны быть UID'ы, посмотри на оф паре, как в УИДы переделали мб...

dageavtobusnick replied:
Викрефы работают на uid у нас. Это просто обертка. Не путать с REF

littleboobs commented on code/game/gamemodes/cult/blood_magic.dm:
```diff
@@ -10,21 +10,21 @@
-			var/order = pix_X + spells.Find(B) * 31
-			B.button.screen_loc = "[screen_loc_X[1]]:[order],[screen_loc_Y[1]]:[screen_loc_Y[2]]"
-			B.button.moved = B.button.screen_loc
+// /datum/action/innate/cult/blood_magic/override_location()
+// 	button.ordered = FALSE
+// 	button.screen_loc = DEFAULT_BLOODSPELLS
+// 	button.moved = DEFAULT_BLOODSPELLS
+
+// /datum/action/innate/cult/blood_magic/proc/Positioning()
+// 	var/list/screen_loc_split = splittext(button.screen_loc, ",")
+// 	var/list/screen_loc_X = splittext(screen_loc_split[1], ":")
+// 	var/list/screen_loc_Y = splittext(screen_loc_split[2], ":")
+// 	var/pix_X = text2num(screen_loc_X[2])
+// 	for(var/datum/action/innate/cult/blood_spell/B in spells)
+// 		if(B.button.locked)
+// 			var/order = pix_X + spells.Find(B) * 31
+// 			B.button.screen_loc = "[screen_loc_X[1]]:[order],[screen_loc_Y[1]]:[screen_loc_Y[2]]"
+// 			B.button.moved = B.button.screen_loc
```
Такие закомменченные куски нужны? Насколько помню, это 1 в 1 копипаститься должно, вроде...

vovakons replied:
а, это культяпки, до них еще руки не дошли, в процессе

vovakons replied:
закомментил так как у меня в моменте было больше тыщи ошибок и я немного подзаебся фиксить все сразу и закомментил большие участки кода

littleboobs commented on code/modules/mob/dead/observer/observer.dm:
```diff
@@ -693,11 +693,11 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
-		if(istype(act.button, /atom/movable/screen/movable/action_button/hide_toggle) || (act in src.actions))
-			continue
-		client.screen |= act.button
+	// Отображение элементов интерфейса (vakons: нахуй надо)
+	// for(var/datum/action/act in mob_eye.actions)
+		//if(istype(act.button, /atom/movable/screen/movable/action_button/hide_toggle) || (act in src.actions))
+			// continue
+		//client.screen |= act.button
```
))

vovakons replied:
пригорело чутка)

littleboobs reviewed (commented):
Герой, но много закомменченного кода, который если не ты, то вряд ли кто-то другой, а если и другой, то придётся вникать. Надеюсь добьёшь экшены 🙏

dageavtobusnick commented on code/datums/components/_component.dm:
```diff
@@ -206,6 +206,11 @@
 			looked_up += src
 
+/// Registers multiple signals to the same proc.
+/datum/proc/RegisterSignals(datum/target, list/signal_types, proctype, override = FALSE)
```
Посмотри внимательно на прок регистер сигнал у нас.

vovakons committed:
register signals in actions fix

Исправлена кривая регистрация сигналов в новом action.

vovakons committed:
refactor drag and drop logic and fix move actions

Рефактор прока MouseDrop, теперь идет через специальный прок
mouse_drop_dragged.
Исправлен баг с перемещением экшн кнопок.

vovakons committed:
Merge branch 'master220' into actions-refactor

vovakons committed:
actions palette port

Портирована фича для скрытия ненужных экшн кнопок.
Фикс обновления активных иконок.

vovakons committed:
implant actions fix

vovakons committed:
linter errors fix

vovakons committed:
linter errors fix 2

vovakons committed:
linter errors fix 3

vovakons committed:
linter errors fix 4

vovakons commented:
Ну вроде как большая часть сделана. Нужны тесты для поиска багов, на локалке потыкал почти все экшны какие смог найти, поправил баги. Нужен тестмерж на пару дней чтобы собрать список багов для фиксов.

vovakons committed:
blood cult spells position

Исправлено позиционирование спеллов кровавого культа (снизу посередине).

vovakons committed:
hostile elite actions fix

Исправление жкшнов элиты лаваленда.

vovakons committed:
admin tickets action button tickets count

Вернул количество открытых тикетов в экшн кнопке админхелпа и
менторхелпа.

vovakons committed:
observer actions fix

Исправлены экшны призраков.
Исправлены экшны админов и менторов.

vovakons committed:
pulse demon action name change logic fix

Исправлена логика изменения имени экшн кнопки у пульс-демона.

vovakons committed:
advanced iem actions charge count

vovakons committed:
strip mobs bugfix

vovakons committed:
Merge branch 'master220' into actions-refactor

vovakons committed:
mob strip fix

dageavtobusnick committed:
Merge branch 'master220' into actions-refactor

vovakons committed:
Merge branch 'master220' into actions-refactor

vovakons committed:
blood cult spirit realm actions fix

Теперь в форме духа культист крови может общаться и проверять цель
культа.

vovakons committed:
cleave attack action fix

Исправление экшн кнопки для переключения клив атаки.

vovakons committed:
accessory actions fix

Исправление жкшн кнопок для акксесуаров навешиваемых на одежду:
- кобура
- разгрузка
- голобейдж
- нательные камеры

vovakons committed:
clockworkers enchant actions

Исправлены экшны зачарования предметов для культа ратвара.

vovakons committed:
actions alt trigger logic

Использование контрол клик для альтернативного действия экшн кнопки.
Фиксит заклинание взгляда вотчера, теперь на контрол клик меняет режим.

littleboobs commented on code/datums/actions/item_action.dm:
```diff
@@ -444,11 +444,11 @@
+		if(target_obj.loc == owner)
 			return TRUE
-		if(istype(target_mob.loc, /obj/item/clothing/under) && target_mob.loc.loc == owner)
+		if(istype(target_obj.loc, /obj/item/clothing/under) && target_obj.loc.loc == owner)
```
`target_obj.loc.loc` так и должно быть?

vovakons replied:
да, `target.loc` это одежда на которую повесили акссесуар, а `target.loc.loc` это пользователь на котором одета одежда

littleboobs replied:
эво как

vovakons committed:
shadowlings actions buttons bug fix

Исправлен баг с поломкой худа при возвышении теней.

vovakons committed:
Merge branch 'master220' into actions-refactor

vovakons committed:
gunlight action fix

dageavtobusnick committed:
Merge branch 'master220' into actions-refactor

dageavtobusnick committed:
ninja fastfix

vovakons committed:
Merge branch 'master220' into actions-refactor

vovakons committed:
pill bottle and patch pack drag use fix

Исправлено поведение при перетаскивании контейнера для пилюль и
контейнера патчей, при перетаскивании на себя применяет все содержимое
на себя, при перетаскивании на пол - опустошает как раньше.

vovakons committed:
give and targeting spell fix

Исправлен click interceptor, теперь нормально возвращает результат
операции, позволяет переопределить клик нормально.

vovakons committed:
change mouse pointer on use spell

Изменение курсора при использовании спеллов требующих цель.

vovakons committed:
active spell overlay

Добавлен оверлей для активного спелла который требует таргетинга.

vovakons committed:
mime silence action fix and research scanner action icon

Исправление иконки для исследовательского анализатора.
Удаление второй кнопки обета молчания у мима от маски.

vovakons committed:
actions observe for ghosts

Добавлен показ экшн кнопок для призраков.

vovakons committed:
matter eat power fix

исправлен бесконечный кд у гена пожирания материи.

vovakons committed:
ninja actions backgrounds fix

Исправление бага с исчезанием бг у экшнов при включении рига.

vovakons committed:
taperoll remove bugfix

Исправление активации предмета когда убираешь скотч с него.

vovakons committed:
Merge branch 'master220' into actions-refactor

vovakons commented:
я честно хз че там сломалось, ошибку не показывает, линтер не трогал

vovakons committed:
remove empty lines for linter
